framework/src/main/java/org/tron/core/Wallet.java

```diff
@@ -4126,11 +4126,12 @@ public byte[] getShieldedContractScalingFactor(byte[] contractAddress)
     TransactionExtention.Builder trxExtBuilder = TransactionExtention.newBuilder();
     Return.Builder retBuilder = Return.newBuilder();
     TransactionExtention trxExt;
+    Transaction trx;
 
     try {
       TransactionCapsule trxCap = createTransactionCapsule(trigger,
           ContractType.TriggerSmartContract);
-      Transaction trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
+      trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
```

Comment:
Looks similar to `isShieldedTRC20NoteSpent`, Are no fixes needed for it?

Reply:
Yes, you are careful and right. I fix it also.